db/data_listeners.hh

```diff
@@ -128,8 +129,8 @@ class toppartitions_data_listener : public data_listener, public weakly_referenc
     friend class toppartitions_query;
 
     database& _db;
-    sstring _ks;
-    sstring _cf;
+    std::unordered_set<std::tuple<sstring, sstring>, utils::tuple_hash> _table_filters;
```

Comment:
that would work, another option is:
std::unordered_map<std::string,std::unordered_set<sstring>> that would cover both options (i.e. for keyspace without table)

Reply:
if there are ever more types of filters we can switch to that, for now I feel like this is nice and straightforward

Reply:
I ment combine the table and keyspace/table to a single case, but this would work